test/open-report.js

```diff
@@ -0,0 +1,77 @@
+import test from 'ava';
+import proxyquire from 'proxyquire';
+
+const openReport = proxyquire('../lib/open-report', {
+	'open-editor': files => files
+});
+
+const createReport = ({results = [], errorCount = 0, warningCount = 0} = {}) => ({
+	results, errorCount, warningCount
+});
+
+const createResult = ({errorCount = 0, warningCount = 0, filePath = '', messages = []} = {}) => ({
+	errorCount, warningCount, filePath, messages
+});
+
+const createMessage = ({line = 1, column = 10} = {}) => ({
+	line, column
+});
+
+test('opens nothing when there are no errors nor warnings', t => {
+	const expected = openReport(createReport());
+	const actual = [];
+
+	t.deepEqual(actual, expected);
+});
+
+test('only opens errors if there are errors and warnings', t => {
+	const report = createReport({
+		errorCount: 1,
+		warningCount: 2,
+		results: [
+			createResult({errorCount: 7, filePath: 'seven-errors.js', messages: [createMessage({line: 7, column: 17})]}),
+			createResult({warningCount: 3, filePath: 'three-warnings.js', messages: [createMessage({line: 3, column: 13})]}),
+			createResult({errorCount: 1, filePath: 'one-error.js', messages: [createMessage()]}),
+			createResult({warningCount: 1, filePath: 'one-warning.js', messages: [createMessage()]})
+		]
+	});
```

Comment:
I've pushed my changes using fixtures. 
If its easier for you to make the changes instead of reviewing it after every change, you can go ahead (not that I don't want to do it, in fact I'm glad with your feedback!).